Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserModel.cpp

```diff
@@ -260,34 +260,78 @@ namespace AzToolsFramework
 
                 if (Utils::FromMimeData(data, entries))
                 {
-                    for (auto entry : entries)
+                    if (entries.empty())
                     {
-                        using namespace AZ::IO;
-                        Path fromPath;
-                        Path toPath;
-                        bool isFolder{ true };
- 
-                        if (entry && (entry->RTTI_IsTypeOf(SourceAssetBrowserEntry::RTTI_Type())))
+                        return false;
+                    }
+
+                    if (entries.size() > 1)
+                    {
+                        for (auto assetEntry : entries)
                         {
-                            fromPath = entry->GetFullPath();
-                            PathView filename = fromPath.Filename();
-                            toPath = item->GetFullPath();
-                            toPath /= filename;
-                            isFolder = false;
+                            if (assetEntry->GetEntryType() == AssetBrowserEntry::AssetEntryType::Folder)
+                            {
+                                return false;
+                            }
                         }
-                        else
+                    }
+                    bool isFolder = entries[0]->GetEntryType() == AssetBrowserEntry::AssetEntryType::Folder;
+                    if (isFolder && AssetBrowserViewUtils::IsEngineOrProjectFolder(entries[0]->GetFullPath()))
+                    {
+                        return false;
+                    }
+                    using namespace AzFramework::AssetSystem;
+                    EntryTypeFilter* foldersFilter = new EntryTypeFilter();
+                    foldersFilter->SetEntryType(AssetBrowserEntry::AssetEntryType::Folder);
+
+                    AZStd::string folderPath = item->GetFullPath();
+                    bool connectedToAssetProcessor = false;
+                    AzFramework::AssetSystemRequestBus::BroadcastResult(
+                        connectedToAssetProcessor, &AzFramework::AssetSystemRequestBus::Events::AssetProcessorIsReady);
```

Comment:
Is it possible to get to this point without the AP being ready? Doesn't the asset browser model have to query the AP database to get its entries?

Reply:
It actually is possible to get here without the AP being ready. If the AP is turned off after the AB model is already loaded, the entries will still appear and we shouldn't be able to perform any file operations on them.